core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt

```diff
@@ -0,0 +1,1508 @@
+    WebView(activity).destroy() // Workaround for buggy webViews see #710
+    handleLocaleCheck()
+    activity.setSupportActionBar(toolbar)
+    actionBar = activity.supportActionBar
+    initHideBackToTopTimer()
+    initTabCallback()
+    toolbar.setOnTouchListener(toolbarSwipeTouchListener)
+    loadDrawerViews()
+    tableDrawerRight =
+      tableDrawerRightContainer.getHeaderView(0).findViewById(R.id.right_drawer_list)
+    addFileReader()
+    setupTabsAdapter()
+    setTableDrawerInfo()
+    setTabListener()
+    compatCallback = CompatFindActionModeCallback(activity)
+    setUpTTS()
+    setupDocumentParser()
+    loadPrefs()
+    updateTitle()
+    handleIntentExtras(activity.intent)
+    tab_switcher_recycler_view!!.adapter = tabsAdapter
+    ItemTouchHelper(tabCallback!!).attachToRecyclerView(tab_switcher_recycler_view)
+
+    // Only check intent on first start of activity. Otherwise the intents will enter infinite loops
+    // when "Don't keep activities" is on.
+    if (savedInstanceState == null) {
+      handleIntentActions(activity.intent)
+    }
+    bottom_toolbar_arrow_back.setOnClickListener { goBack() }
+    bottom_toolbar_arrow_forward.setOnClickListener { goForward() }
+    bottom_toolbar_toc.setOnClickListener { openToc() }
+    activity_main_fullscreen_button.setOnClickListener { closeFullScreen() }
+    tab_switcher_close_all_tabs.setOnClickListener { closeAllTabs() }
+    bottom_toolbar_bookmark.setOnClickListener { toggleBookmark() }
+    bottom_toolbar_bookmark.setOnLongClickListener { goToBookmarks() }
+    bottom_toolbar_home.setOnClickListener { openMainPage() }
+    activity_main_back_to_top_fab.setOnClickListener { backToTop() }
+    activity_main_button_pause_tts.setOnClickListener { pauseTts() }
+    activity_main_button_stop_tts.setOnClickListener { stopTts() }
+  }
+
+  private fun initTabCallback() {
+    tabCallback = object : ItemTouchHelper.Callback() {
+      override fun getMovementFlags(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder
+      ): Int {
+        return makeMovementFlags(
+          0,
+          ItemTouchHelper.UP or ItemTouchHelper.DOWN
+        )
+      }
+
+      override fun onChildDraw(
+        c: Canvas,
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        dX: Float,
+        dY: Float,
+        actionState: Int,
+        isCurrentlyActive: Boolean
+      ) {
+        super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive)
+        viewHolder.itemView.alpha = 1 - abs(dY) / viewHolder.itemView.measuredHeight
+      }
+
+      override fun onMove(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        target: RecyclerView.ViewHolder
+      ): Boolean = false
+
+      override fun onSwiped(
+        viewHolder: RecyclerView.ViewHolder,
+        direction: Int
+      ) {
+        closeTab(viewHolder.adapterPosition)
+      }
```

Comment:
Definitely expression body here will look awesome

Reply:
we should use expression functions when we return a useful value,  using an expression function to return a `Unit` feels incorrect to me